libs/wingsdk/src/cloud/api.ts

```diff
@@ -432,6 +430,40 @@ export class Api extends Resource {
     };
   }
 
+  /**
+   * Checks if two given paths are siblings.
+   * @param pathA
+   * @param pathB
+   * @returns A boolean value indicating if provided paths are siblings.
+   * @internal
+   */
+
+  protected _arePathsSiblings(pathA: string, pathB: string): boolean {
+    const partsA = pathA.split("/");
+    const partsB = pathB.split("/");
+
+    for (let i = 0; i < partsA.length; i++) {
+      const partA = partsA[i];
+      const partB = partsB[i];
+      if (
+        (!partA.match(/^:.+?$/) || !partB.match(/^:.+?$/)) &&
+        partA[i] !== partB[i]
+      ) {
+        return false;
+      }
+
+      if (
+        partA.match(/^:.+?$/) &&
+        partB.match(/^:.+?$/) &&
+        partA[i] !== partB[i]
+      ) {
+        return true;
+      }
+    }
+
+    return false;
+  }
```

Comment:
I've added some tests for the sibling paths in the `libs/wingsdk/test/target-sim/api.test.ts` - since it was already a part of the api class, and I didn't want to take it out if it's only used there- especially since we don't have any of those in the other resources, and I didn't know where to locate the test file